packages/browser/src/extensions/conversations/external/index.tsx

```diff
@@ -0,0 +1,616 @@
+     * The user continues their conversation seamlessly - widget_session_id stays the same.
+     * No migration needed since tickets are keyed by widget_session_id, not distinct_id.
+     * Backend will update ticket.distinct_id for Person linking on the next message.
+     */
+    private _handleDistinctIdChange(oldDistinctId: string, newDistinctId: string): void {
+        // No migration needed - widget_session_id stays the same
+        // The user keeps access to their ticket because the widget_session_id hasn't changed
+        logger.info('User identified, conversation continues with same widget_session_id', {
+            ticketId: this._currentTicketId,
+            widgetSessionId: this._widgetSessionId,
+            oldDistinctId,
+            newDistinctId,
+        })
+
+        // Track the identity change
+        this._posthog.capture('$conversations_identity_changed', {
+            hadExistingTicket: !!this._currentTicketId,
+        })
+    }
+
+    /**
+     * Enable/show the widget (button and chat panel)
+     */
+    enable(): void {
+        this._widgetRef?.show()
+    }
+
+    /**
+     * Disable/hide the widget completely (button and chat panel)
+     */
+    disable(): void {
+        this._widgetRef?.hide()
+    }
+
+    /**
+     * Send a message programmatically (internal use)
+     */
+    sendMessage(message: string): void {
+        this._handleSendMessage(message)
+    }
+
+    /**
+     * Clean up the widget
+     */
+    destroy(): void {
+        this._stopPolling()
+
+        // Unsubscribe from identify events
+        if (this._unsubscribeIdentifyListener) {
+            this._unsubscribeIdentifyListener()
+            this._unsubscribeIdentifyListener = null
+        }
+
+        if (this._containerElement) {
+            render(null, this._containerElement)
+            this._containerElement.remove()
+            this._containerElement = null
+        }
+
+        this._widgetRef = null
+        logger.info('Widget destroyed')
+    }
+
+    /**
+     * Reset all conversation data and destroy the widget.
+     * Called on posthog.reset() to start fresh.
+     */
+    reset(): void {
+        // Clear all persisted conversation data
+        this._persistence.clearAll()
+
+        // Reset local state
+        this._currentTicketId = null
+        this._lastMessageTimestamp = null
+        this._unreadCount = 0
+
+        // Destroy the widget
+        this.destroy()
+
+        logger.info('Conversations reset')
+    }
+
+    /**
+     * Render the widget to the DOM
+     */
+    private _renderWidget(initialState: ConversationsWidgetState, initialUserTraits: UserProvidedTraits | null): void {
+        if (!document) {
+            logger.info('Conversations widget not rendered: Document not available')
+            return
+        }
+
+        // Create container if it doesn't exist
+        let container = document.getElementById(WIDGET_CONTAINER_ID) as HTMLDivElement
```

Comment:
another potential follow-up you could register a dom loaded callback here if the document isn't ready yet, so that if there's a race this does get called  
  
(i think it's maybe unlikely we get to the point of executing this code before the DOM is loaded though so :shrug:)